software/lazydogpile.livecodescript

```diff
@@ -0,0 +1,954 @@
+end loadButtMouseUp
+
+on saveAsButtMouseUp
+   -- get the file path for this stack
+   put the effective filename of this stack into tPath
+   set the itemDelimiter to slash
+   -- if this is a standalone application running on a Mac, find the path to the .app file
+   if the platform = "MacOS" and the environment = "standalone application" then
+      repeat until last item of tPath contains ".app"
+         delete last item of tPath
+      end repeat
+   end if
+   -- if not on a Mac, remove the last item to get the path to the containing folder
+   --   delete last item of tPath
+   
+   -- shouldn't put project folders into /data, it's inside the github dev repository
+   -- documents folder makes sense as a default, but allow for anywhere
+   put specialFolderPath("documents") into tDocsPath
+   put sMainShort[sNowDoc][2]["title"] into tCaseName
+   if tCaseName is empty then put "[blank]" into tCaseName
+   ask file "New project name (the case title is: " & tCaseName& ")"  with tDocsPath
+   if the result is not "cancel" then
+      put it into tReq["newPath"]
+      put jsonFromArray(sMain[sNowDoc]) into tReq["fileData"]
+      put saveNewProject(tReq) into tFullName
+      put "saved" into sMain[sNowDoc][0]["saveStatus"]
+      put tFullName into sMain[sNowDoc][0]["filepath"]
+      put item -1 of tFullName into tForShow
+      replace ".how" with "" in tForShow
+      put tForShow into field "filenameLabel" stack "HowstrBeta"
+   end if -- the result is not cancel
+   
+   --   put "data" into item -1 of tPath -- files go into the data folder
+   --   -- if the case node already has a name in it, suggest that
+   --   put sMainShort[sNowDoc][2]["title"] into tCaseName
+   --   ask ("name this file (do not include a file extension)" & cr & tPath) with tCaseName
+   --   if the result is not "cancel" then
+   --      put it into tFileName
+   --      put jsonFromArray(sMain[sNowDoc]) into tResult
+   --      -- each project can have lots of associated files, so the project file gets its own folder
+   --      create folder tPath & "/" & tFileName
+   --      put tPath & "/" & tFileName & "/" & tFileName & ".how" into tFullName
+   --      put tResult into URL ("file:" & tFullName)
+   --   put "saved" into sMain[sNowDoc][0]["saveStatus"]
+   --   put tFullName into sMain[sNowDoc][0]["filepath"]
+   --   put item -1 of tFullName into tForShow
+   --   replace ".how" with "" in tForShow
+   --   put tForShow into field "filenameLabel" stack "HowstrBeta"
+   --   end if
+end saveAsButtMouseUp
+
+on iconCloseMouseUp
+   put "on closeStackRequest" into xX["write"];    devLog xX
+   if sMain[sNowDoc][0]["saveStatus"] is "unsaved" then
+      answer "Close without saving?" with "Close" or "Cancel"
+      if it is "cancel" then exit iconCloseMouseUp
+   end if
+   quit
+end iconCloseMouseUp
+
+on closeStackRequest
+   -- for whatever reason this only gets sent by the window decoration (at least in Windows where I tested it)
```

Comment:
@matthewmaier Actually because you aren't passing this is going to block any stack from closing unless messages are locked so if you don't delete you need to add `pass closeStackRequest`